src/model.py

```diff
@@ -2172,22 +2173,18 @@ def forward(self, batch):
             for task in self.subtasks:
                 task_output = task_inputs[task]
 
-                if f"cnn_{stack}" in self.components[task].keys():
-                    if split_cnn_tasks is not None and task in split_cnn_tasks: 
-                        cnn_outputs[task] = torch.cat(
-                            [
-                                cnn(task_output)
-                                for cnn in self.components[task][f"cnn_{stack}"]
-                            ],
-                            dim=1
-                        )
-                    else:
-                        cnn_outputs[task] = self.components[task][f"cnn_{stack}"](task_output)
+                if split_cnn_tasks is not None and task in split_cnn_tasks: 
+                    cnn_outputs[task] = torch.cat(
+                        [
+                            cnn(task_output)
+                            for cnn in self.components[task][f"cnn_{stack}"]
+                        ],
+                        dim=1
+                    )
+                else:
+                    cnn_outputs[task] = self.components[task][f"cnn_{stack}"](task_output)
```

Comment:
No longer a need to check that `cnn_0` is in this task's component keys. Every task will have it, as long as stack>0. This is when the empty sequential passes pay off. 

When stack==0, this code piece is never reached.